SerialPrograms/Source/PokemonSV/Programs/PokemonSV_Navigation.cpp

```diff
@@ -870,11 +925,20 @@ bool attempt_fly_to_overlapping_flypoint(
     ConsoleHandle& console, 
     BotBaseContext& context
 ){
-    open_map_from_overworld(info, console, context);
-    context.wait_for_all_requests();
-    pbf_press_button(context, BUTTON_ZL, 40, 100);
+    while (true){
+        try {
+            open_map_from_overworld(info, console, context, false);
+            context.wait_for_all_requests();
+            pbf_press_button(context, BUTTON_ZL, 40, 100);
+
+            return fly_to_overworld_from_map(info, console, context, true);            
+
+        }catch (UnexpectedBattleException& e){
+            (void) e;

```

Comment:
If you omit the parameter name in the catch declaration, you don't need to do this.

`}catch (UnexpectedBattleException&){`